CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -650,10 +650,12 @@ protected function processParams($commentStart, $commentEnd)
                     } else if (count($typeNames) === 1) {
                         // Check type hint for array and custom type.
                         $suggestedTypeHint = '';
-                        if (strpos($suggestedName, 'array') !== false) {
+                        if (strpos($suggestedName, 'array') !== false || substr($suggestedName, -2) === '[]') {
```

Comment:
This does cover basic `ClassName[]` format, but doesn't cover more complex ones, like `integer[]|string` or `(classA|classB)|(classC[])`. I personally don't use such complex types in DocBlocks, but PhpDoc supports them.

Supporting them might be bad for performance reasons (because you need to parse whole type string).